From 12f0521bbb3f046db8c7854364135d8247cf982e Mon Sep 17 00:00:00 2001
From: Armin Novak <armin.novak@thincast.com>
Date: Fri, 29 Jan 2016 17:04:12 +0100
Subject: [PATCH] Disabled library versioning.

---
 Makefile.shared | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile.shared b/Makefile.shared
index e753f44..5cffcc2 100644
--- a/Makefile.shared
+++ b/Makefile.shared
@@ -81,7 +81,6 @@ CALC_VERSIONS=	\
 		prev=""; \
 		for v in `echo "$(LIBVERSION) $(LIBCOMPATVERSIONS)" | cut -d';' -f1`; do \
 			SHLIB_SOVER_NODOT=$$v; \
-			SHLIB_SOVER=.$$v; \
 			if [ -n "$$prev" ]; then \
 				SHLIB_COMPAT="$$SHLIB_COMPAT .$$prev"; \
 			fi; \
-- 
2.1.4

